be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs

```diff
@@ -163,12 +163,12 @@ public async Task<PagedResults<IEnumerable<OrderServiceModel>>> GetAsync(GetOrde
 
             if (model.IsSeller is false)
             {
-                orders = orders.Where(x => x.SellerId == model.OrganisationId.Value);
+                orders = orders.Where(x => x.SellerId == model.OrganisationId.Value || x.OrderItems.Any(y => y.ExternalReference.StartsWith(model.SearchTerm)) || x.Id.ToString() == model.SearchTerm);
             }
 
             if (!string.IsNullOrWhiteSpace(model.SearchTerm))
             {
-                orders = orders.Where(x => x.ClientName == model.SearchTerm || x.OrderItems.Any(y => y.ExternalReference == model.SearchTerm));
+                orders = orders.Where(x => x.ClientName.StartsWith(model.SearchTerm) || x.OrderItems.Any(y => y.ExternalReference.StartsWith(model.SearchTerm)) || x.Id.ToString() == model.SearchTerm);
```

Comment:
suggestion: Please add ToLowercaseInvariant() to ignore the captial and lowercase in search.